### PR TITLE
Decode blobIdHeader in RequestPath for named blob case

### PR DESCRIPTION
**Pull request by mikeyang923**
Decode blobIdHeader in RequestPath for named blob case

### CODE REVIEW

File: ambry-api/src/main/java/com/github/ambry/rest/RequestPath.java [hunk 1]
```diff
@@ -55,6 +60,16 @@ public static RequestPath parse(RestRequest restRequest, List<String> prefixesTo
...
+      Object requestPath = restRequest.getArgs().get(InternalKeys.REQUEST_PATH);
+      if (requestPath != null && ((RequestPath) requestPath).matchesOperation(Operations.NAMED_BLOB)) {
+        try {
+          blobIdHeader = URLDecoder.decode(blobIdHeader, StandardCharsets.UTF_8.name());
+        } catch (UnsupportedEncodingException e) {
+          throw new RestServiceException(
+              "Fail to decode the blobIdHeader value: " + blobIdHeader + " for signed named blob: " + e.getMessage(),
+              RestServiceErrorCode.UnsupportedEncoding);
+        }
+      }
```
[the remark this answers is not in the document]
Comment (mikeyang923) [reply ambry-api/src/main/java/com/github/ambry/rest/RequestPath.java]: I see, just updated, can you pls take another look?

File: ambry-api/src/main/java/com/github/ambry/rest/RequestPath.java [hunk 2]
```diff
@@ -55,6 +60,13 @@ public static RequestPath parse(RestRequest restRequest, List<String> prefixesTo
       // If blob id is specified in the rest request header and it has cluster name prefix, we remove its prefix, subResource
       // and use this id (pure blob id with no prefix or subResource) to update "x-ambry-blob-id" header. The purpose is
       // to ensure IdConverter and IdSigningService can receive a valid blob id and correctly identify it.
+      try {
+        blobIdHeader = URLDecoder.decode(blobIdHeader, StandardCharsets.UTF_8.name());
+      } catch (UnsupportedEncodingException e) {
+        throw new RestServiceException(
+            "Fail to decode the blobIdHeader value: " + blobIdHeader + " for signed named blob: " + e.getMessage(),
```
Comment (SophieGuo410) [inline ambry-api/src/main/java/com/github/ambry/rest/RequestPath.java:67]: Can you please update the error message? the blobId is not just for signed named blob.

Comment (justinlin-linkedin) [reply ambry-api/src/main/java/com/github/ambry/rest/RequestPath.java:67]: Please add some test cases for this code.

**Comment by mikeyang923** [comment]
I will close this for now, since this PR currently only has simple few lines change, and i may be slow to address comments due to the work in grid team.

**Pull request closed without merging**